### ISSUE TITLE
IP has a flat cost

### ISSUE THREAD

**Issue by yamatt** [yamatt/fly-minecraft-server#1]
This doesn't scale to zero because we're attaching to an IP, and the IP has a flat cost. Not much, but not zero.

**Comment by lambdachad** [comment]
Possible to attach domain to it? Or use the provided domain?

**Comment by yamatt** [comment]
Hey @svelterust thanks for your interest in this.

IIRC when you allocate an IP to a machine, the domain that fly.io sets up uses that IP.

Otherwise it's easy enough to configure your own domain provider to use that IP.

**Comment by lambdachad** [comment]
Am I missing something obvious? As far as I know, it's the dedicated IPv4 addresses that cost 2$/month each. The default shared IPv4 shouldn't cost anything.

**Comment by yamatt** [comment]
That is true for web apps. However, when you're using TCP endpoints, which is what Minecraft uses, you need to pay for an IP.

If you could run Minecraft over HTTP it would be free 😁
